src/rules/completedDocsRule.ts

```diff
@@ -387,24 +357,37 @@ class CompletedDocsWalker extends Lint.ProgramAwareRuleWalker {
             return;
         }
 
-        const requirement = this.requirements.get(nodeType);
-        if (!requirement || !requirement.shouldNodeBeDocumented(node)) {
+        const exclusions = this.exclusionsMap.get(nodeType);
+        if (!exclusions) {
             return;
         }
 
-        const symbol = this.getTypeChecker().getSymbolAtLocation(node.name);
-        if (!symbol) {
-            return;
+        for (const exclusion of exclusions) {
+            if (exclusion.excludes(node)) {
+                return;
+            }
         }
 
-        const comments = symbol.getDocumentationComment();
-        this.checkComments(node, nodeType, comments);
+        if (this.declarationHasNoComments(node, this.getTypeChecker())) {
+            this.addDocumentationFailure(node, nodeType);
+        }
     }
 
-    private checkComments(node: ts.Declaration, nodeDescriptor: string, comments: ts.SymbolDisplayPart[]) {
-        if (comments.map((comment: ts.SymbolDisplayPart) => comment.text).join("").trim() === "") {
-            this.addDocumentationFailure(node, nodeDescriptor);
+    private getCommentLines(node: ts.Declaration, typeChecker: ts.TypeChecker) {
+        if (node.name === undefined) {
+            return [];
         }
+
+        const symbol = typeChecker.getSymbolAtLocation(node.name);
+        if (symbol === undefined) {
+            return [];
+        }
+
+        return symbol.getDocumentationComment();
```

Comment:
starting from ts@2.1 you don't necessarily need the typechecker for that, see my other comment above